src/internal/m365/collection/site/handlers.go

```diff
@@ -20,3 +20,32 @@ type getItemByIDer interface {
 type getItemser interface {
 	GetItems(ctx context.Context, cc api.CallConfig) ([]models.Listable, error)
 }
+
+type restoreHandler interface {
+	PostLister
+	PostListItemer
+	DeleteLister
+}
+
+type PostLister interface {
+	PostList(
+		ctx context.Context,
+		listName string,
+		oldListByteArray []byte,
```

Comment:
naming nits:
1. `old` implies that there's a `newListByteArray` somewhere.  I don't think that's the case.
2. this is a `slice`, not an `array`.

```suggestion
		listData []byte,
```